bip-0322.mediawiki

```diff
@@ -13,119 +13,99 @@
 
 == Abstract ==
 
-A standard for interoperable generic signed messages based on the Bitcoin Script format.
-
-== Background ==
-
-* Assume two actors, a prover <code>P</code> and a verifier <code>V</code>.
-* <code>P</code> wants to prove that they own the private key <code>k</code> associated with a given address <code>A</code> (which in turn is derived from the pubkey <code>kG</code>).
-* Let <code>V</code> generate a message <code>M</code> and hand this to <code>P</code>.
-* <code>P</code> generates a signature <code>S</code> by signing the message <code>M</code> using <code>k</code>. Given <code>S</code>, <code>V</code> can prove that <code>P</code> has the private key associated with <code>A</code>.
-
-The astute reader will notice that the above is missing a critical part, namely the pubkey <code>kG</code>, without which the verifier cannot actually verify the message. The current message signing standard solves this via a cryptographic trick, wherein the signature <code>S</code> above is a special "recoverable signature" type. Given the message <code>M</code> and the signature <code>S</code>, it is then possible to recover the pubkey <code>kG</code>. The system thus derives the address for the pubkey <code>kG</code>, and if it does not match <code>A</code>, the proof is deemed invalid.
-
-While this is a neat trick, it unnecessarily restricts and complicates the message signing mechanism; for instance, it is currently not possible to sign a message for a P2SH address, because there is no pubkey to recover from the resulting signature.
+A standard for interoperable signed messages based on the Bitcoin Script format, either for proving fund availability, or committing to a message as a recipient of an invoice address.
```

Comment:
A custodian might allow customers to sign messages with their designated deposit addresses, but the customers may not control the keys and just receive the promise of being able to withdraw equivalent amount in the future (potentially from completely unrelated UTXO).

It seems to me that "committing to a message as the intended recipient of the funds to be sent to the invoice address" would be more fitting description

Reply:
Excellent point! I like the rephrasing, but think it could be put a bit more succinctly: "committing to a message as the intended recipient of funds sent to the invoice address".

Reply:
Changed!